common/src/main/java/com/box/l10n/mojito/openai/OpenAIClient.java

```diff
@@ -236,71 +251,58 @@ public interface Message {
       String name();
     }
 
-    public record SystemMessage(String role, String content, String name) implements Message {
-
-      public static class SystemMessageBuilder {
```

Comment:
@maallen I'm curious about that change. Was there an issue with having the multiple Message types? Some types have different fields (eg. assistant), so I thought have it typed was better.

Reply:
Hi @ja-openai!

I added functionality to allow sending message context history with a prompt, initially I added an assistant message builder for building assistant messages but it was felt that code was being duplicated to repeatedly set the content, so I removed into a single builder which defaults to role type `user` if not provided.

https://github.com/pinterest/mojito/pull/122#discussion_r1631298034